### PR TITLE
Add CDPVault_TypeB

### PR DESCRIPTION
**Pull request by A6957E**

### CODE REVIEW

**Review by d1dff2 — APPROVED** [review]

File: src/CDPVault_TypeB.sol [hunk 1]
```diff
@@ -0,0 +1,431 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {IERC20} from "openzeppelin/contracts/token/ERC20/IERC20.sol";
```
Comment (F2E4D1) [inline src/CDPVault_TypeB.sol:4]: Align import style

File: src/interfaces/ICDPVaultUnwinder.sol [hunk 2]
```diff
@@ -18,7 +18,7 @@ interface ICDPVaultUnwinder {
     
     function AUCTION_DURATION() external view returns (uint256);
 
-    function vault() external view returns (ICDPVault);
+    function vault() external view returns (ICDPVault_TypeB);
```
Comment (F2E4D1) [inline src/interfaces/ICDPVaultUnwinder.sol:21]: generalize for ICDPVault

File: src/CDPVault_TypeB_Factory.sol [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            uint256(constants.targetUtilizationRatio)
+            | (uint256(constants.maxUtilizationRatio) << 64)
+            | (uint256(constants.minInterestRate - WAD) << 128)
+            | (uint256(constants.maxInterestRate - WAD) << 168)
+            | (uint256(constants.targetInterestRate - WAD) << 208);
+        rebateParams = uint256(constants.rebateRate) | (uint256(constants.maxRebate) << 128);
+    }
+    
+
+    function creditWithholder() external returns (address) {
```
Comment (F2E4D1) [inline src/CDPVault_TypeB_Factory.sol:145]: This doesn't seem right

File: src/CDPVault_TypeB.sol (hunk 1 above)
Comment (d1dff2) [reply src/CDPVault_TypeB.sol:4]: done

File: src/interfaces/ICDPVaultUnwinder.sol (hunk 2 above)
Comment (d1dff2) [reply src/interfaces/ICDPVaultUnwinder.sol:21]: I propose we fix this in a follow up PR

**Pull request merged**